src/main/java/servlets/BillingConfig.java

```diff
@@ -14,36 +14,65 @@
+  private static final String REDIRECT_READFILE = "/index.html";
+
+  public static final String VENDOR_ID = "vendor-id";
+  public static final String ACCOUNT_ID = "account-id";
+  public static final String LEGACY_CUSTOMER_ID = "legacy-customer-id";
+  public static final String NEXT_GEN_CUSTOMER_ID = "next-gen-customer-id";
+  public static final String LEGACY_ACCOUNT_ID = "legacy-account-id";
+  public static final String NEXT_GEN_ACCOUNT_ID = "next-gen-account-id";
+  public static final String CURRENCY_CODE = "currency-code";
+  public static final String DIRECTION = "direction";
+  public static final String ENTITY = "entity";
+  public static final String MATCHING_MODE = "matching-mode";
+  public static final String PRODUCT_ACCOUNT_KEY = "product-account-key";
+  public static final String AGGREGATION_MODE = "aggregation-mode";
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // GET method --> returns entire billing config as JSON
     // return null if vendor ID doesn't exist
-    response.setContentType("text/html;");
-    response.getWriter().println(
-            "{ \"legacy_customer_id\": \"STRING\", " +
-            "\"next_gen_customer_id\": \"INT\", \"accounts\": [ { \"legacy_account_id\": " +
-            "\"STRING\", \"next_gen_customer_id\": \"INT\", \"settlement_attributes\": " +
-            "{ \"currency_code\": \"STRING\", \"direction\": \"STRING\", \"entity\": " +
-            "\"STRING\" }, \"settlement_config\": { \"matching_mode\": \"STRING\" }, " +
-            "\"product_account_key\": \"STRING\", \"aggregation_mode\": \"STRING\" }, " +
-            "{ \"legacy_account_id\": \"STRING\", \"next_gen_customer_id\": \"INT\"," +
-            " \"settlement_attributes\": { \"currency_code\": \"STRING\", \"direction\": " +
-            "\"STRING\", \"entity\": \"STRING\" }, \"settlement_config\": { \"matching_mode\": " +
-            "\"STRING\" }, \"product_account_key\": \"STRING\", \"aggregation_mode\": " +
-            "\"STRING\" } ] }"
-    );
+    String vendorID = request.getParameter(VENDOR_ID);
+    String accountID = request.getParameter(ACCOUNT_ID);
+    JsonConverter json = new JsonConverter();
+    String configText = "";
+
+    if (json.getConfig(vendorID, false) != null) {
+      configText = json.getConfig(vendorID, false);
+    } else {
+      configText = "Error finding " + vendorID + "'s configuration";
+    }
+
+    response.setContentType(CONTENT_TYPE_TEXT_HTML);
+    response.getWriter().println(configText);
+    response.sendRedirect(REDIRECT_READFILE);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // POST method --> overwrites the existing configuration or creates a new one
+    // currently assuming there is only one account --> add this to the form @Vincent?
+    // TODO: can the vendor ID be appended to the request as a query string @vincent?
```

Comment:
Maybe? Query stuff is more on the java side of things iirc

Reply:
handling the query string is done by the java servlet, but I think the best way for you to share the vendor and account ID will be with a query string, since they'll have been selected through a different form. I can show you some code how I've done it before

Reply:
Yeah, for sure. I think I'm gonna move the formatting on the front-end to the search form.